core/src/main/java/org/apache/gravitino/storage/relational/service/FilesetMetaService.java

```diff
@@ -125,13 +119,31 @@ public FilesetEntity getFilesetByIdentifier(NameIdentifier identifier) {
   public List<FilesetEntity> listFilesetsByNamespace(Namespace namespace) {
     NamespaceUtil.checkFileset(namespace);
 
-    Long schemaId =
-        EntityIdService.getEntityId(
-            NameIdentifier.of(namespace.levels()), Entity.EntityType.SCHEMA);
-
-    List<FilesetPO> filesetPOs =
-        SessionUtils.getWithoutCommit(
-            FilesetMetaMapper.class, mapper -> mapper.listFilesetPOsBySchemaId(schemaId));
+    List<FilesetPO> filesetPOs;
+    if (GravitinoEnv.getInstance().config().get(Configs.CACHE_ENABLED)) {
+      Long schemaId =
+          EntityIdService.getEntityId(
+              NameIdentifier.of(namespace.levels()), Entity.EntityType.SCHEMA);
+      filesetPOs =
+          SessionUtils.getWithoutCommit(
+              FilesetMetaMapper.class, mapper -> mapper.listFilesetPOsBySchemaId(schemaId));
+    } else {
+      String[] namespaceLevels = namespace.levels();
+      filesetPOs =
+          SessionUtils.getWithoutCommit(
+              FilesetMetaMapper.class,
+              mapper ->
+                  mapper.listFilesetPOsByFullQualifiedName(
+                      namespaceLevels[0], namespaceLevels[1], namespaceLevels[2]));
+      if (filesetPOs.isEmpty() || filesetPOs.get(0).getSchemaId() == null) {
```

Comment:
`filesetPOs` empty should be expected, which means there's no fileset, it cannot be mixed with SchemaNotFound issue, right?

Reply:
I have used a left join to fetch `schemaPO`, and as long as the schema exists, the `filesetPos` will never be null or empty

Reply:
A list of filesetPOs and the `filesetId` of all are null.